runtime/flang/miscsup_com.c

```diff
@@ -4544,6 +4544,11 @@ ENTF90(NEARESTX, nearestx)(__REAL4_T f, __LOG_T sign)
       ++x.i;
     else
       --x.i;
+  } else if ((x.i & 0x7FFFFFFF) == 0x7F800000) {
+    if (~sign && (x.i & 0xFFFFFFFF) == 0x7F800000)
```

Comment:
Is there any reason for testing `(x.i & 0x7FFFFFFF) == 0x7F800000` just before testing `(x.i & 0xFFFFFFFF) == 0x7F800000`? It does not seem necessary to group the two conditions in one `else if` block.

Reply:
Also, is it correct to use `~sign` instead of `!(sign & GET_DIST_MASK_LOG)`?